File: 836.rectangle-overlap.py
class Solution:
    def isRectangleOverlap(self, rec1, rec2):
        """
        :type rec1: List[int]
        :type rec2: List[int]
        :rtype: bool
        """
        x11, y11, x21, y21 = rec1
        rec2_x1, rec2_y1, rec2_x2, rec2_y2 = rec2
        return not (x11 >= rec2_x2 or x21 <= rec2_x1 or y11 >= rec2_y2 or y21 <= rec2_y1)

File: 836.rectangle-overlap.cpp
class Solution {
public:
    bool isRectangleOverlap(vector<int>& rec1, vector<int>& rec2) {
        return !(rec1[0] >= rec2[2]    // rec1完全在rec2右侧
                 || rec1[2] <= rec2[0] // rec1完全在rec2左侧
                 || rec1[1] >= rec2[3] // rec1完全在rec2上方
                 || rec1[3] <= rec2[1] // rec1完全在rec2下方
                 || rec1[0] == rec1[2] // rec1为一条竖线
                 || rec1[1] == rec1[3] // rec1为一条横线
                 || rec2[0] == rec2[2] // rec2为一条竖线
                 || rec2[1] == rec2[3] // rec2为一条横线
            );
    }
};